Client/Ecosia/UI/PageAction/PageActionMenuCell.swift

```diff
@@ -0,0 +1,225 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+
+final class PageActionMenuCell: UITableViewCell {
+    
+    struct UX {
+        
+        /// The cell's identifier
+        static let cellIdentifier = String(describing: PageActionMenuCell.self)
+        
+        /// The corner radius to apply to the cells
+        /// depending on their position
+        static let cornerRadius: CGFloat = 10.0
+        
+        /// The cell's left / right padding
+        static let padding: CGFloat = 16.0
+        
+        /// The separator height of the `PageActionMenuCell`'s `separatorView`
+        static let separatorHeight: CGFloat = 1.0
+        
+        /// This `enum` serves the purpose of checking the cells' position
+        /// within a section
+        enum Position {
+            /// Only one cell is rendered within a section
+            case solo
+            /// The cell is the first of its section
+            case first
+            /// The cell is between the first and last of its section
+            case middle
+            /// The cell is the last of its section
+            case last
+        }
+    }
+    
+    private weak var badge: UIView?
+    private weak var badgeLabel: UILabel?
+    private var separatorView: UIView?
+    
+    /// The cell's position, starting from `middle` as preferred value
+    private var position: UX.Position = .middle
+    
+    /// The eligible cell positions within the Table View
+    /// to add the separator view
+    private let separatorCellsPositions: [UX.Position] = [.middle, .first]
+    
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: .subtitle, reuseIdentifier: reuseIdentifier)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        adjustPadding()
+        setTableViewCellCorners()
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        separatorView = nil
+        textLabel?.text = nil
+        imageView?.image = nil
+    }
+}
+
+extension PageActionMenuCell {
+    
+    /// Determines the TableView Cell's position based at a given Index Path
+    ///
+    /// - Parameters:
+    ///    - indexPath: The Table View's index path
+    ///    - actions: The array of `PhotonRowActions` utilized as Data Source
+    func determineTableViewCellPositionAt(_ indexPath: IndexPath, forActions actions: [[PhotonRowActions]]) {
```

Comment:
Just a thought: This method looks like a great candidate to be unit tested as it contains quite some logic. It also looks like a great candidate to be stateless (e.g. being a class function) and return Position instead of mutating it on the cell itself. That way it's also more elegant to unit test it. What do you think?

Reply:
Eventually kept the `func` as an instance one and added the `@discardableResult` to make it testable. Commenting to close the loop here.